tlatools/org.lamport.tlatools/src/tlc2/tool/fp/LongArray.java

```diff
@@ -1,310 +1,209 @@
-		rangeCheck(position2);
-		final long tmp = get(position1);
-		set(position1, get(position2));
-		set(position2, tmp);
+	public final void swap(final long position1, final long position2) throws IndexOutOfBoundsException {
+		final long tmp = this.get(position1);
+		this.set(position1, this.get(position2));
+		this.set(position2, tmp);
 	}
 	
-	/*
-	 * Variant of swap that uses copyMemory. This implementation - suprisingly - is
-	 * *not* faster compared to swap above (see LongArrayBenchmark).
+	/**
+	 * Returns the capacity of this array.
+	 *
+	 * @return The capacity of this array.
 	 */
-	void swapCopy(final long position1, final long position2) {
-		final long tmp = unsafe.getAddress(log2phy(position1));
-		unsafe.copyMemory(log2phy(position2), log2phy(position1), 8L);
-		unsafe.putAddress(log2phy(position2), tmp);
-	}
-
-	
-    /**
-     * Returns the number of elements in this array.
-     *
-     * @return the number of elements in this array
-     */
 	public final long size() {
-		return length;
+		return this.capacity;
 	}
 	
-	/* (non-Javadoc)
+	/**
+	 * Dumps the contents of this array into a string.
+	 *
 	 * @see java.lang.Object#toString()
+	 * @return A string representation of the array.
 	 */
-	public String toString() {
-		return toString(0L, length - 1L);
+	public String toString() throws IndexOutOfBoundsException {
+		assert this.capacity <= (long)Integer.MAX_VALUE : "LongArray is too long to convert to a string.";
+		return toString(0L, this.capacity - 1L);
 	}
 
-	public String toString(long start, long end) {
-        long iMax = end;
-        if (iMax == -1L) {
-        	return "[]";
-        }
+	/**
+	 * Dumps a subsequence of this array to a string.
+	 *
+	 * @param start The index to begin reading.
+	 * @param end The index to end reading (inclusive).
+	 * @return A string representation of the array.
+	 */
+	public String toString(long start, long end) throws IndexOutOfBoundsException {
+		this.rangeCheck(start);
+		this.rangeCheck(end);
+		assert start <= end : "Start must not be higher than end.";
+		assert (end - start) <= (long)Integer.MAX_VALUE : "LongArray subsequence is too long to convert to a string.";
+		long iMax = end;
+		if (iMax == -1L) {
+			return "[]";
+		}
 
-        final StringBuilder b = new StringBuilder();
-        b.append('[');
-        for (long i = start; ; i++) {
-            final long lng = get(i);
-            if (lng == 0L) {
-            	b.append("e");
-            } else {
-            	b.append(lng);
-            }
-            if (i == iMax) {
-            	return b.append(']').toString();
-            }
-            b.append(", ");
-        }
-	}
-	
-	public static void main(final String[] args) throws IOException {
-		final long elements = 1L << Integer.valueOf(args[0]);
-		System.out.format("Allocating LongArray with %,d elements.\n", elements);
-		final LongArray longArray = new LongArray(elements);
-		longArray.zeroMemory();
+		final StringBuilder b = new StringBuilder();
```

Comment:
Why remove the main?

Reply:
Why not? Nobody uses it, it's dead code.

Reply:
Please don't unilaterally decide what is dead code.

Reply:
Do you use it?

Reply:
Maybe this can be moved to a new class in `test-benchmark/`?